x-pack/plugins/reporting/server/browsers/chromium/driver_factory/test.ts

```diff
@@ -0,0 +1,159 @@
+  '--disable-backgrounding-occluded-windows',
+  '--disable-breakpad',
+  '--disable-client-side-phishing-detection',
+  '--disable-component-extensions-with-background-pages',
+  '--disable-default-apps',
+  '--disable-dev-shm-usage',
+  '--disable-extensions',
+  '--disable-features=TranslateUI',
+  '--disable-hang-monitor',
+  '--disable-ipc-flooding-protection',
+  '--disable-popup-blocking',
+  '--disable-prompt-on-repost',
+  '--disable-renderer-backgrounding',
+  '--disable-sync',
+  '--force-color-profile=srgb',
+  '--metrics-recording-only',
+  '--no-first-run',
+  '--enable-automation',
+  '--password-store=basic',
+  '--use-mock-keychain',
+  '--remote-debugging-port=0',
+  '--headless',
+];
+
+interface BrowserDiagnosticPayload {
+  help: string[];
+  success: boolean;
+  logs: string;
+}
+
+export const browserTest = (
+  core: ReportingCore,
+  logger: LevelLogger,
+  flags: string[] = []
+): Promise<BrowserDiagnosticPayload> => {
+  const config = core.getConfig();
+  const proxy = config.get('capture', 'browser', 'chromium', 'proxy');
+  const userDataDir = mkdtempSync(join(tmpdir(), 'chromium-'));
+  const binaryPath = getBinaryPath();
+  const kbnArgs = args({
+    userDataDir,
+    viewport: { width: 800, height: 600 },
+    disableSandbox: true,
+    proxy,
+  });
+  const finalArgs = uniq([...defaultArgs, ...kbnArgs, ...flags]);
+  let browserLogs = '';
+  let closed = false;
+
+  const browserProcess = spawn(binaryPath, finalArgs, {
+    // On non-windows platforms, `detached: true` makes child process a
+    // leader of a new process group, making it possible to kill child
+    // process tree with `.kill(-pid)` command. @see
+    // https://nodejs.org/api/child_process.html#child_process_options_detached
+    detached: process.platform !== 'win32',
+  });
+
+  return new Promise((resolve) => {
+    const cleanup = async (err?: Error) => {
+      if (closed) return;
+      if (err) {
+        logger.error(`Browser didn't start or didn't bind to the local port in time`);
+        logger.error(err);
+      }
+      const help: BrowserDiagnosticPayload['help'] = [];
+      closed = true;
+      clearTimeout(timer);
+
+      // Sometimes logs with significant issues can creep in after browser has bound it's remote
+      // debugging port, so let them flow through before SIGKILL is sent.
+      await sleep(browserCoolDown);
+
+      browserProcess.removeAllListeners();
+      rl.removeAllListeners();
+      rl.close();
+
+      for (const knownIssue of BROWSER_TEST_COMMON_ISSUES.keys()) {
+        const helpText = BROWSER_TEST_COMMON_ISSUES.get(knownIssue);
+        if (browserLogs.includes(knownIssue) && helpText) {
+          logger.warn(helpText);
+          help.push(helpText);
+        }
+      }
+
+      if (browserProcess && browserProcess.pid && !browserProcess.killed) {
+        try {
+          browserProcess.kill('SIGKILL');
+        } catch (error) {
+          logger.warn(
```

Comment:
It's a little unclear based on the comments inside `browserProcess` variable declaration, but is it assumed this error will happen on Windows?